aws/resource_aws_ecrpublic_repository_policy_test.go

```diff
@@ -0,0 +1,296 @@
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSEcrPublicRepositoryPolicy(randString),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSEcrPublicRepositoryPolicyExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "policy"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccAWSEcrPublicRepositoryPolicyUpdated(randString),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSEcrPublicRepositoryPolicyExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "policy"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSEcrPublicRepositoryPolicy_iam(t *testing.T) {
+	randString := acctest.RandString(10)
+	resourceName := "aws_ecrpublic_repository_policy.default"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAwsEcrPublic(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsEcrPublicRepositoryPolicyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSEcrPublicRepositoryPolicyWithIAMRole(randString),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSEcrPublicRepositoryPolicyExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "policy"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccAWSEcrPublicRepositoryPolicyWithIAMRoleUpdated(randString),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSEcrPublicRepositoryPolicyExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "policy"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAwsEcrPublicRepositoryPolicyDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).ecrconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_ecrpublic_repository_policy" {
+			continue
+		}
+
+		_, err := conn.GetRepositoryPolicy(&ecr.GetRepositoryPolicyInput{
+			RegistryId:     aws.String(rs.Primary.Attributes["registry_id"]),
+			RepositoryName: aws.String(rs.Primary.Attributes["repository"]),
+		})
+		if err != nil {
+			if ecrerr, ok := err.(awserr.Error); ok && ecrerr.Code() == "RepositoryNotFoundException" {
+				return nil
+			}
+			return err
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckAWSEcrPublicRepositoryPolicyExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		_, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		return nil
+	}
+}
+
+func testAccAWSEcrPublicRepositoryPolicy(randString string) string {
+	return fmt.Sprintf(`
+resource "aws_ecrpublic_repository" "foo" {
+  repository_name = "tf-acc-test-ecr-%s"
+}
+
+resource "aws_ecrpublic_repository_policy" "default" {
+  repository = aws_ecrpublic_repository.foo.repository_name
```

Comment:
For this and other test configurations:

```suggestion
func testAccAWSEcrPublicRepositoryPolicy(rName string) string {
	return fmt.Sprintf(`
resource "aws_ecrpublic_repository" "test" {
  repository_name = %[1]q
}

resource "aws_ecrpublic_repository_policy" "test" {
  repository_name = aws_ecrpublic_repository.test.repository_name
```